synapse/handlers/oidc_handler.py

```diff
@@ -906,20 +908,25 @@ async def _map_userinfo_to_user(
         localpart = map_username_to_mxid_localpart(attributes["localpart"])
 
         user_id = UserID(localpart, self._hostname)
-        if await self._datastore.get_users_by_id_case_insensitive(user_id.to_string()):
-            # This mxid is taken
-            raise MappingException(
-                "mxid '{}' is already taken".format(user_id.to_string())
-            )
-
-        # It's the first time this user is logging in and the mapped mxid was
-        # not taken, register the user
-        registered_user_id = await self._registration_handler.register_user(
-            localpart=localpart,
-            default_display_name=attributes["display_name"],
-            user_agent_ips=(user_agent, ip_address),
+        matches = await self._datastore.get_users_by_id_case_insensitive(
+            user_id.to_string()
         )
-
+        if matches:
+            if self._allow_existing_users:
+                registered_user_id = next(iter(matches))
```

Comment:
Can matches have multiple items? Should we still error in that case? It feels arbitrary to choose the first one. At the very least I think we should log something.

Reply:
I think there can't be more than one item as the same (case insensitive) usernames are not allowed. This is also why we need `get_users_by_id_case_insensitive`.

https://github.com/matrix-org/synapse/blob/efb6b6629c78409251f61857f2bfe6c2f8f8fb8d/synapse/handlers/register.py#L111-L116

Maybe we should log something if there're more than one, to show that something goes wrong? But it's a little strange. If we do, we should also do this in many other places.

Reply:
Well, I find `auth` considers more. But I still can't understand why there can be multiple matches. Maybe for backward compatibility？
https://github.com/matrix-org/synapse/blob/efb6b6629c78409251f61857f2bfe6c2f8f8fb8d/synapse/handlers/auth.py#L753-L758